main.c

```diff
@@ -455,8 +460,21 @@ void action_write(const char *filename, minipro_handle_t *handle, device_t *devi
 	}
 }
 
+/* TODO: Optimize this */
+int get_id(device_t *device) {	
+	minipro_handle_t *handle = minipro_open(device);
+	minipro_begin_transaction(handle);
+	unsigned int get_chip_id = minipro_get_chip_id(handle);
+	minipro_end_transaction(handle);
+	printf("0x%02x\n", get_chip_id);
+	exit(1);
+}
+
 int main(int argc, char **argv) {
 	parse_cmdline(argc, argv);
+  if(cmdopts.get_id) {
```

Comment:
You don't seem to be following the coding style/indentation of the rest of the code.